packages/@ember/application/lib/application.js

```diff
@@ -844,7 +844,7 @@ const Application = Engine.extend({
       _loaded.application = undefined;
     }
 
-    if (this._applicationInstances.size) {
+    if (this._applicationInstances && this._applicationInstances.size) {
```

Comment:
Hmm, why isn't `this._applicationInstances` defined (it is setup in `init`, which should be before anything else here has accessed it)?

Reply:
I am not sure why I needed this, it's been like 7 weeks. Anyway, reverted this change.